lib/rules/id-match.js

```diff
@@ -78,7 +78,21 @@ module.exports = {
         const IMPORT_TYPES = new Set(["ImportSpecifier", "ImportNamespaceSpecifier", "ImportDefaultSpecifier"]);
 
         /**
-         * Checks if a string matches the provided pattern
+         * Checks whether the given node represents a reference to a global variable that is not declared in the source code.
+         * These identifiers will be allowed, as it is assumed that user has no control over the names of external global variables.
+         * @param {ASTNode} node `Identifier` node to check.
+         * @returns {boolean} `true` if the node is a reference to a global variable.
+         */
+        function isReferenceToGlobalVariable(node) {
+            const scope = context.getScope();
+            const variable = scope.set.get(node.name);
+
+            return variable && variable.defs.length === 0 &&
+                variable.references.some(ref => ref.identifier === node);
+        }
+
+        /**
+         * Checks if a string matches the provided pattern and is not a global reference
```

Comment:
```suggestion
         * Checks if a string matches the provided pattern
```

This function still only checks if a string matches the provided pattern.

Reply:
leftover from old commit.